openedx/features/ucsd_features/ecommerce/tests/test_utils.py

```diff
@@ -2,29 +2,113 @@
 Tests for ecommerce utils
 """
 import httpretty
+import mock
 
 from django.conf import settings
+from django.test.utils import override_settings
 from mock import patch
 
 from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
 from xmodule.modulestore.tests.factories import CourseFactory
 from student.tests.factories import UserFactory
 
 from openedx.features.ucsd_features.ecommerce.utils import is_user_eligible_for_discount
+from openedx.features.ucsd_features.ecommerce.tests.utils import make_ecommerce_url
 
 
 class UCSDFeaturesEcommerceUtilsTests(ModuleStoreTestCase):
 
-    def setUpClass(self):
-        super(UCSDFeaturesEcommerceUtilsTests, self).setUpClass()
-        self.course = CourseFactory.create()
-
     def setUp(self):
         super(UCSDFeaturesEcommerceUtilsTests, self).setUp()
         self.user = UserFactory()
+        self.course = CourseFactory.create()
 
     def test_is_user_eligible_for_discount_with_disabled_feature(self):
-        request = mock.MagicMock()
+        features = settings.FEATURES
+        features['ENABLE_GEOGRAPHIC_DISCOUNTS'] = False
```

Comment:
I think we can use 
`from django.test import override_settings`
@override_settings(
        FEATURES={}
    )
it will restrict change of a variable to just that specific test case

Reply:
that will override all the feature flags. I simply want to change the one affecting my test case.

Reply:
Also, overriding with `with` also restricts the change to the scope below as well.

Reply:
when we are doing `features['ENABLE_GEOGRAPHIC_DISCOUNTS'] = False` its changing the value globally, which I don't think we want here to be done.
secondly, just asking, can't we do something like:
@override_settings(
FEATURES['variable']='value'
)

Reply:
You are right about the "global" change. We can fix that by making a copy of `FEATURES` dict
Regarding your code suggestion, I don't think we can do that. there will be syntax error. 
https://stackoverflow.com/questions/24214636/django-override-settings-does-not-allow-dictionary

Reply:
Ok, I was just concerned about the global change if you can handle that by copying dict that would be great.
No need to do this, just saying, We can use copy() function to do something like:
@override_settings(
FEATURES=settings.FEATURES.copy().update({
     'a':'b',
     'c':'d'
})
)

Reply:
Cool

Reply:
Didn't work. `update` actually updates the caller dict and doesn't return anything.